lgc/builder/BuilderImpl.h

```diff
@@ -294,8 +294,8 @@ class DescBuilder : virtual public BuilderImplBase {
                                    const llvm::Twine &instName) override final;
 
   // Create a pointer to a descriptor.
-  llvm::Value *CreateGetDescPtr(ResourceNodeType descType, unsigned descSet, unsigned binding,
-                                const llvm::Twine &instName) override final;
+  llvm::Value *CreateGetDescPtr(ResourceNodeType concreteType, unsigned descSet, unsigned binding,
```

Comment:
It's tempting to just add arguments at the end. However, let's be careful about how we setup our APIs, and I *think* that this would be slightly easier to use if the arguments were:
```
CreateGetDescPtr(concreteType, abstractType, descSet, binding, instName)
```
With that order of arguments, the `abstractType` is clearly "closer" to `descSet` and `binding`. So if, six months from now, we remember that there are two types because one is used as a sort of lookup key and the other determines the actual concrete type of descriptor we need, then it is easier to remember that "the type that is closer to the set and binding, which are used as lookup keys, is the type that is used to lookup". And then by exclusion, the other type must be the concrete one.

I hope that makes sense :)